### PR TITLE
Added petri net node type

### PR DESCRIPTION
**Pull request by robbr48**
New node type for petri net connections.

Name: NodePetriNet
Variables: State, Flow
Nice variable names: s, q

New components:
PetriNetInterfaceC, PetriNetInterfaceQ

### CODE REVIEW

File: HopsanCore/include/Nodes.h [hunk 1]
```diff
@@ -471,6 +471,27 @@ class NodeModelica : public Node
...
 
+//! @brief A petri net node
+//! @ingroup NodePetriNet
+class NodePetriNet : public Node
+{
+public:
+    //! @brief The data variable indexes, DataLength is used internally
+    //! @ingroup NodePetriNet
+    enum DataIndexEnumT {State, Flow, DataLength};
+    enum DataIndexEnumOldT {STATE, FLOW};
```
Comment (peterNordin) [inline HopsanCore/include/Nodes.h]: The DataIndexEnumOldT is not neede for new nodes, better to force users to use the new non-uppercase versions

File: componentLibraries/defaultLibrary/Connectivity/PetriNetInterfaceC.hpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#ifndef PETRINETINTERFACEC_HPP_INCLUDED
+#define PETRINETINTERFACEC_HPP_INCLUDED
+
+#include "ComponentEssentials.h"
+
+namespace hopsan {
+
+    //!
+    //! @brief
+    //! @ingroup MechanicalComponents
```
Comment (peterNordin) [inline componentLibraries/defaultLibrary/Connectivity/PetriNetInterfaceC.hpp]: Is this really a mechanical component?

File: componentLibraries/defaultLibrary/Connectivity/PetriNetInterfaceQ.hpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+#ifndef PETRINETINTERFACEQ_HPP_INCLUDED
+#define PETRINETINTERFACEQ_HPP_INCLUDED
+
+#include "ComponentEssentials.h"
+
+namespace hopsan {
+
+    //!
+    //! @brief
+    //! @ingroup MechanicalComponents
```
Comment (peterNordin) [inline componentLibraries/defaultLibrary/Connectivity/PetriNetInterfaceQ.hpp]: Mechanical ?

File: HopsanCore/include/Nodes.h (hunk 1 above)
Comment (robbr48) [reply HopsanCore/include/Nodes.h]: Ok, I was unsure of this, but it sounds reasonable.

File: componentLibraries/defaultLibrary/Connectivity/PetriNetInterfaceQ.hpp (hunk 3 above)
Comment (robbr48) [reply componentLibraries/defaultLibrary/Connectivity/PetriNetInterfaceQ.hpp]: I added an `InterfaceComponents` group for all interface components (they were in fact all using `MechanicalComponents`).

**Pull request merged**